.openshift-ci/sensor_versions.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+"""
+Returns the latest patches of the last 4 major versions
+"""
+
+import sys
+import re
+import subprocess
+
+def isReleaseTagGitCLI(version):
+    return bool(re.search(r"^\d+\.\d+\.\d+$", version))
+
+def filterGitCLITags(rawtags):
+    filteredtags = []
+    for t in rawtags:
+        if isReleaseTagGitCLI(t):
+            filteredtags.append(t)
+    return set(filteredtags)
+
+def cliOutputToTags(stdoutput):
+    separated = stdoutput.decode(encoding="utf-8").splitlines()
+    return separated
+
+def transformTagsToNumbers(tags):
+    numTags = []
+    for t in tags:
+        numTags.append(splitVersion(t))
+    return sorted(numTags)
+
+def splitVersion(version):
+    digits = re.search(r"(\d+)\.(\d+)\.\D*(\d+)", version)
+    return int(digits.group(1)), int(digits.group(2)), int(digits.group(3))
+
+def getLatestTags(tags, num_versions):
+    numericaltags = transformTagsToNumbers(tags)
+    _,y,_ = splitVersion(makeImageTag())
```

Comment:
Is `makeImageTag` command necessary here? Is there a scenario where the result of this command will be _lower_ than the latest release? This will be frequently run on PRs, which are on the latest major version anyway. 

The only case where this will be lower is when we're building patches for older releases. But in this case we might end up testing sensor versions that are already outside the support window. E.g.: patching 3.69.x and running compatibility tests for `3.68.x`, `3.67.x` and `3.66.x`. 

I'd prefer to simply run the tests with the latest 4 released versions of sensor, regardless of the tag from `make image`.

Reply:
I think it makes a lot of sense to test it this way, eg. ensuring that it's the last 4 major versions from current central version down that we test. We may run into unintended behavior if lets say we test for central `3.68.x` but sensor in the tests is in version `3.72.x`. And while it's true that we do not support older releases, at the time that an older version was released the compatibility for versions that were within the compatibility window at that time should be expected.

It's questionable whether these tests should / would be run on older centrals at all, but if they are I prefer the current behavior.